Include/Rocket/Core/StringBase.inl

```diff
@@ -26,42 +26,42 @@
  */
 
 template< typename T >
-StringBase< T >::StringBase() : value((T*)local_buffer), buffer_size(LOCAL_BUFFER_SIZE), length(0), hash(0)
+StringBase< T >::StringBase() : value(LocalBuffer()), buffer_size(LOCAL_BUFFER_SIZE), length(0), hash(0)
```

Comment:
No value added, just complexity.  Rather than restate this comment in multiple places, this applies to pretty much every typecast removed.